ruuvitag_sensor/ruuvi.py

```diff
@@ -156,17 +156,22 @@ def _get_ruuvitag_datas(macs=[], search_duratio_sec=None, run_flag=RunFlag(), bt
             if not run_flag.running:
                 data_iter.send(StopIteration)
                 break
-            # Check MAC whitelist
-            if macs and not ble_data[0] in macs:
+            # Check MAC whitelist if advertised MAC available
+            if ble_data[0] and macs and not ble_data[0] in macs:
                 continue
+
             (data_format, data) = DataFormats.convert_data(ble_data[1])
             # Check that encoded data is valid RuuviTag data and it is sensor data
-            # If data is not valid RuuviTag data add MAC to blacklist
+            # If data is not valid RuuviTag data add MAC to blacklist if MAC is available
             if data is not None:
-                state = get_decoder(data_format).decode_data(data)
-                if state is not None:
-                    yield (ble_data[0], state)
+                decoded = get_decoder(data_format).decode_data(data)
+                if decoded is not None:
+                    # Check whitelist using MAC from decoded data if advertised MAC is not available
+                    if not ble_data[0] and decoded['mac'] and macs and not decoded['mac'] in macs:
```

Comment:
Note that `mac` in ruuvi pay load comes in a format without `:` like `112c6a1e593d`

`':'.join(mac_address[i:i+2] for i in range(0,12,2))` should make it standard like the others.

Reply:
Thanks for pointing it out. Made fix :+1: